lab-kyle/lib/parse-body.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+module.exports = function(req, res, cowsay, helper, callback) {
+  req.body = '';
+
+  req.on('data', function(data) {
+    req.body += data;
+    let parsedBody = JSON.parse(req.body);
+
+    if(parsedBody.text) {
+      helper(200, parsedBody.text, parsedBody.animal);
+    } else {
+      helper(400, 'bad request\ntry: localhost:3000/cowsay?text=howdy');
+    }
+  });
```

Comment:
You'll also want a `req.on('end'...` which should ultimately be the place where you helper function grabs the completed `req.body` and hands it back out of the parser.